crates/wasm-smith/src/lib.rs

```diff
@@ -584,29 +577,34 @@ where
 
```

Comment:
This is an existing problem that I just noticed, but these choices should probably not be loop invariant anymore, since we want to respect the config's max functions and globals.

Reply:
Interesting -- this is assuming that the imported functions and globals count against the limits, yes? My assumption had been that the limits should denote the number of functions or globals defined in the module itself; this simplifies the import generation and also removes the particular conflict between the minimum import count and the maximum allowed element (func/global/mem/table) counts below.